trestle/core/commands/author/ssp.py

```diff
@@ -0,0 +1,364 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Create ssp from catalog and profile."""
+
+import argparse
+import logging
+import pathlib
+from typing import Dict, List, Optional, Set, Union
+
+import trestle.core.generators as gens
+import trestle.oscal.catalog as cat
+import trestle.oscal.profile as prof
+import trestle.oscal.ssp as ossp
+import trestle.utils.fs as fs
+import trestle.utils.log as log
+from trestle.core.commands.author.common import AuthorCommonCommand
+from trestle.core.markdown_validator import MarkdownValidator
+from trestle.utils.load_distributed import load_distributed
+from trestle.utils.md_writer import MDWriter
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+class ControlHandle():
+    """Convenience class for handling controls as member of a group."""
+
+    def __init__(self, group_id: str, group_title: str, control: cat.Control):
+        """Initialize the control handle."""
+        self.group_id = group_id
+        self.group_title = group_title
+        self.control = control
+
+
+class SSP(AuthorCommonCommand):
+    """Create SSP in markdown form from Catalog and Profile, or assemble the markdown into a json SSP."""
+
+    name = 'ssp'
+
+    def __init__(self, parser=None, parent=None, name=None, out=None, err=None):
+        """Initialize the class."""
+        super().__init__(parser, parent, name, out, err)
+        self._param_dict: Dict[str, str] = None
+        self._md_file: MDWriter = None
+        self._alters: List[prof.Alter] = None
+        self._yaml_header: dict = None
+        self._sections: Dict[str, str] = None
+
+    def _init_arguments(self) -> None:
+        file_help_str = 'Name of the profile file'
+        self.add_argument('-f', '--file', help=file_help_str, required=True, type=str)
+        yaml_help_str = 'Path to the yaml header file'
+        self.add_argument('-yh', '--yaml-header', help=yaml_help_str, required=True, type=str)
+        sections_help_str = 'Comma separated list of section:alias pairs for sections to output'
+        self.add_argument('-s', '--sections', help=sections_help_str, required=False, type=str)
+        task_help_str = 'Name of the generated ssp markdown folder'
+        self.add_argument('-o', '--output', help=task_help_str, required=True, type=str)
+        verbose_help_str = 'Display verbose output'
+        self.add_argument('-v', '--verbose', help=verbose_help_str, required=False, action='count', default=0)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        log.set_log_level_from_args(args)
+        trestle_root = fs.get_trestle_project_root(pathlib.Path.cwd())
+        if not trestle_root:
```

Comment:
A side note - the initialisation function that was provided in AuthorCommonCommand probably should be further abstracted as this is a common check across many commands. (e.g. have a version that does not rely on class names in a lower level base class.